build.sbt

```diff
@@ -1,10 +1,29 @@
+import sbt.Keys.libraryDependencies
 
-crossScalaVersions := Seq("2.11.12", "2.12.12")
+lazy val CommonSettings = Seq(
+  version := (
+    if (git.gitCurrentBranch.value == "main") {
+      "latest"
+    } else {
+      git.gitCurrentBranch.value
+    }) + "-SNAPSHOT",
+)
 
-organization := "co.datamechanics"
-name := "delight"
-version := "latest-SNAPSHOT"
+lazy val global = project
+  .in(file("."))
+  .aggregate(
+    agent
+  )
+  .settings(
+    publish / skip := true
+  )
 
-libraryDependencies += "org.apache.spark" %% "spark-core" % "2.4.3"
-
-publishTo := sonatypePublishToBundle.value
+lazy val agent = (project in file("agent"))
+  .settings(
+    name := "delight",
```

Comment:
nit: Shouldn't `name` and others be part of `CommonSettings`?

Reply:
We'll soon introduce a new package so putting this in common will soon be useless (this PR is preparation for this)